drivers/i2c/i2c_sf32lb.c

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c) 2025, Qingsong Gou <gouqs@hotmail.com>
+ * Copyright (c) 2025, Haoran Jiang <halfsweet@halfsweet.cn>
+ * Copyright (c) 2025, SiFli Technologies(Nanjing) Co., Ltd
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sifli_sf32lb_i2c
+
+#include "i2c-priv.h"
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control/sf32lb.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(i2c_sf32lb, CONFIG_I2C_LOG_LEVEL);
+
+#include <register.h>
+
+#define I2C_CR    offsetof(I2C_TypeDef, CR)
```

Comment:
Likewise having no name prefix here on I2C_TypeDef reduces readability in my view but I understand there's probably not much to be done about it